exasol_transformers_extension/udfs/models/base_model_udf.py

```diff
@@ -137,6 +138,14 @@ def _check_values_not_null(model_name, bucketfs_conn, sub_dir):
                             f"Found model_name = {model_name}, bucketfs_conn = {bucketfs_conn}, sub_dir = {sub_dir}."
             raise ValueError(error_message)
 
+    def set_work_with_spans(self, model_df: pd.DataFrame):
+        if "span" in model_df.columns:
+            # todo this only gets set after some other things have already happened in udf.
+            #   which means if udf fails before the returned table does not include "token_span" column.
+            #   this results in the return shape of the table potentially being inconsistent.
+            #   do we want an input column specifying "work with spans instead?
+            self.work_with_spans = True
```

Comment:
work_with_spans should be part of the constructor and we configure it in the UDF template

Reply:
default is false